www/js/control/AppStatusModal.tsx

```diff
@@ -371,16 +370,16 @@ const AppStatusModal = ({permitVis, setPermitVis}) => {
      //load when ready
      useEffect(() => {
         if (appConfig && window['device']?.platform) {
-            setPlatform(window['device'].platform.toLowerCase());
-            setOsver(window['device'].version.split(".")[0]);
+            platform.current = window['device'].platform.toLowerCase();
+            osver.current = window['device'].version.split(".")[0];
```

Comment:
I have no objections around this since it works, but for my own edification: I thought that it was not OK in react to set values directly - you needed to call a `set` function instead. But here, you are setting the values directly!?

Is this OK because it is a part of `useEffect`?

Reply:
It's ok because they are `useRef` variables, a change from a `useState` variable because those are updated asynchronously. Since the `platform` and `osver` are set once, and not expected to change, I switched the way they were stored (at Jack's suggestion). The docs I linked in my comments above outline a bit about the difference between `useRef` and `useState`, but my basic understanding is that we use `useState` when the variable controls something in the UI, because we need to re-render when it changes, whereas `useRef` does not trigger a re-render (this is demonstrated in some examples on that docs page). The platform and osver here are not going to change, so we don't need them to be able to trigger a re-render. The asynchronous nature of `useState` was what was causing the blank checklist issue because the platform and osver were not always set before the checks were initialized.